{"updated_at": "2013-05-02T23:26:26Z", "repo": "rails", "created_at": "2013-05-02T23:26:26Z", "owner": "rails", "url": "https://api.github.com/repos/rails/rails/issues/comments/17371032", "id": 17371032, "issue_url": "https://api.github.com/repos/rails/rails/issues/10047", "issue_id": 10047, "user": {"repos_url": "https://api.github.com/users/saki7/repos", "type": "User", "html_url": "https://github.com/saki7", "starred_url": "https://api.github.com/users/saki7/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/saki7/subscriptions", "following_url": "https://api.github.com/users/saki7/following{/other_user}", "gravatar_id": "3bb47dada671eb91ef01a7b9fdf6d254", "organizations_url": "https://api.github.com/users/saki7/orgs", "url": "https://api.github.com/users/saki7", "gists_url": "https://api.github.com/users/saki7/gists{/gist_id}", "avatar_url": "https://2.gravatar.com/avatar/3bb47dada671eb91ef01a7b9fdf6d254?d=https%3A%2F%2Fidenticons.github.com%2F5e3bb4009c71d11f780fabfc20783ccc.png", "id": 1901813, "events_url": "https://api.github.com/users/saki7/events{/privacy}", "login": "saki7", "received_events_url": "https://api.github.com/users/saki7/received_events", "followers_url": "https://api.github.com/users/saki7/followers"}, "body": "Hi, I came from the issue #8573.\r\nI understand about the race conditions.\r\nIMO, raising the DB constraints error / unique validation error (as mentioned by @wangjohn) is the same thing. Adding an unique validation still does not improve usefulness; it just provides you a chance to catch the exception. You must catch it every time you deal with the association.\r\n\r\nSo,\r\n\r\n```ruby\r\nperson.tags << tag unless person.tags.exists?(id: tag.id)\r\n```\r\ncan't this be a simple workaround for those requiring an easy de-duplications?\r\nYou will just get the validation error as usual, if forgot to check existence.", "_id": {"$oid": "5237e397bd3543c15100916b"}, "html_url": "https://github.com/rails/rails/pull/10047#issuecomment-17371032"}
